client/mm/exchange_adaptor_test.go

```diff
@@ -1918,3 +2879,144 @@ func TestExchangeAdaptorTrade(t *testing.T) {
 		runTest(test)
 	}
 }
+
+func TestOrderFeesInUnits(t *testing.T) {
+	type test struct {
+		name      string
+		buyFees   *orderFees
+		sellFees  *orderFees
+		rate      uint64
+		market    *MarketWithHost
+		fiatRates map[uint32]float64
+
+		expectedSellBase  uint64
+		expectedSellQuote uint64
+		expectedBuyBase   uint64
+		expectedBuyQuote  uint64
+	}
+
+	tests := []*test{
+		{
+			name: "dcr/btc",
+			market: &MarketWithHost{
+				BaseID:  42,
+				QuoteID: 0,
+			},
+			buyFees: &orderFees{
+				swap:       5e5,
+				redemption: 1.1e4,
+			},
+			sellFees: &orderFees{
+				swap:       1.085e4,
+				redemption: 4e5,
+			},
+			rate:              5e7,
+			expectedSellBase:  810850,
+			expectedBuyBase:   1011000,
+			expectedSellQuote: 405425,
+			expectedBuyQuote:  505500,
+		},
+		{
+			name: "btc/usdc.eth",
+			market: &MarketWithHost{
+				BaseID:  0,
+				QuoteID: 60001,
+			},
+			buyFees: &orderFees{
+				swap:       1e7,
+				redemption: 4e4,
+			},
+			sellFees: &orderFees{
+				swap:       5e4,
+				redemption: 1.1e7,
+			},
+			fiatRates: map[uint32]float64{
+				60001: 0.99,
+				60:    2300,
+				0:     42999,
+			},
+			rate:              calc.MessageRateAlt(43000, 1e8, 1e6),
+			expectedSellBase:  108838,
+			expectedBuyBase:   93489,
+			expectedSellQuote: 47055555,
+			expectedBuyQuote:  40432323,
+		},
+		{
+			name: "wbtc.polygon/usdc.eth",
+			market: &MarketWithHost{
+				BaseID:  966003,
+				QuoteID: 60001,
+			},
+			buyFees: &orderFees{
+				swap:       1e7,
+				redemption: 4e4,
+			},
+			sellFees: &orderFees{
+				swap:       5e4,
```

Comment:
Noting that when converting from MATIC to WBTC, the converted contribution is 0.
```
5e4 matic / 1e9 * 0.8 / 42500 * 1e8 = 0.09 Sats < 1 Sat
```